include/tscore/Allocator.h

```diff
@@ -142,6 +140,9 @@ template <class C> class ClassAllocator : public Allocator
   void
   free(C *ptr)
   {
+    if (Destruct_on_free) {
```

Comment:
Seems a nice idea. I always struggle with how to destroy/free objects when I use `ClassAllocator`.